### ISSUE TITLE
model.predict throwing constant trend value error for exogenous variable for period=1 and working fine for period>=1

### ISSUE THREAD

**Issue by notdsr** [alkaline-ml/pmdarima#376]
**Question**

```
model = pm.auto_arima(df.iloc[:-20][['Consumption']], exogenous = df.iloc[:-20][['Income']],  start_p=1, start_q=1, max_p=3, max_q=3, seasonal=False, d=0, trace=True,error_action='ignore',  suppress_warnings=True, stepwise=True)
     
model.predict(n_periods=1, return_conf_int=True, exogenous=df.iloc[-20:-19][['Income']])
     
**returns**  ValueError (ValueError: A constant trend was included in the model specification, but the `exog` data already 
         contains a column of constants)
```

     But predicting for 2 periods works totally fine.
 

**Versions (if necessary)**

System:
    python: 3.7.6 (default, Jan  8 2020, 20:23:39) [MSC v.1916 64 bit (AMD64)]
executable: C:\Users\deepa\anaconda3\python.exe
   machine: Windows-10-10.0.18362-SP0

Python dependencies:
        pip: 20.0.2
 setuptools: 45.2.0.post20200210
    sklearn: 0.22.1
statsmodels: 0.12.0rc0
      numpy: 1.18.1
      scipy: 1.4.1
     Cython: 0.29.15
     pandas: 1.0.1
     joblib: 0.14.1
   pmdarima: 1.5.3
​

**Comment by tgsmith61591** [comment]
Thanks for the report, I'll take a look into this

**Comment by tgsmith61591** [comment]
This is related to statsmodels' latest release 0.12.0. It seems like a bug on their end. We might want to pin to statsmodels <0.12.0 for the time being

**Comment by tgsmith61591** [comment]
This was fixed by pinning to statsmodels<0.12 in the latest release: `1.7.1`. Will work on a longer-term fix for the next release cycle

**Comment by notdsr** [comment]
Thanks!!!

**Comment by ChadFulton** [comment]
The underlying issue in Statsmodels should be fixed in today's release of v0.12.1.

**Comment by aaronreidsmith** [comment]
Looks good on 0.12.1 (See #394). Thanks @ChadFulton (and the rest of the statsmodels team)!
